src/classes/job.ts

```
@@ -39,6 +39,7 @@ const logger = debuglog('bull');
 const optsDecodeMap = {
   fpof: 'failParentOnFailure',
   kl: 'keepLogs',
+  rdof: 'removeDependencyOnFailure',
```

since the options are mutually exclusive, we should probably need to throw an exception if the user sets both to true...